### PR TITLE
Bootstrap shell script to support new packages.broadcom.com infrastructure

### LINKED ISSUE #2027: salt-bootstrap scripts broken: Need a new release to work with packages.broadcom.com repos
### Description of Issue/Question

`salt-bootstrap` needs updating to work with the new `packages.broadcom.com` repo hosting, layout, and package management layer of version pinning (apt, dnf), the removal of all Salt versions before Salt 3006 LTS, etc. since the migration (Salt Project Blog post: [Salt Project Package Repository (repo.saltproject.io) Migration and Guidance](https://saltproject.io/blog/salt-project-package-repo-migration-and-guidance/) and [Salt Install Guide Overhaul and salt-bootstrap updates](https://saltproject.io/blog/salt-install-guide-overhaul/)).

The Salt Install Guide has been updated to show how people can install on target systems without `salt-bootstrap` at the moment, and can be used as a reference point for updating `salt-bootstrap` logic:

- https://docs.saltproject.io/salt/install-guide/en/latest/

### Follow-up steps

This also means, upon completion, that documentation in both this repo and the https://github.com/saltstack/salt-install-guide will need updates based on changes in the script behavior.

- https://github.com/saltstack/salt-install-guide/issues/3

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
changes shell bootstrap script to support new packages.broadcom.com infrastructure

### What issues does this PR fix or reference?
Fixes https://github.com/saltstack/salt-bootstrap/issues/2027

### New Behavior
Able to use bootstrap script for Linux with the new infrastructure at packages.broadcom.com

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -2922,24 +2936,45 @@ __install_saltstack_ubuntu_repository() {
...
     # SaltStack's stable Ubuntu repository:
-    SALTSTACK_UBUNTU_URL="${HTTP_VAL}://${_REPO_URL}/${_ONEDIR_DIR}/${__PY_VERSION_REPO}/ubuntu/${UBUNTU_VERSION}/${__REPO_ARCH}/${STABLE_REV}"
-    echo "$__REPO_ARCH_DEB $SALTSTACK_UBUNTU_URL $UBUNTU_CODENAME main" > /etc/apt/sources.list.d/salt.list
-    __apt_key_fetch "$SALTSTACK_UBUNTU_URL/SALT-PROJECT-GPG-PUBKEY-2023.gpg" || return 1
+    ## DGM SALTSTACK_UBUNTU_URL="${HTTP_VAL}://${_REPO_URL}/${_ONEDIR_DIR}/${__PY_VERSION_REPO}/ubuntu/${UBUNTU_VERSION}/${__REPO_ARCH}/${STABLE_REV}"
+    ## DGM echo "$__REPO_ARCH_DEB $SALTSTACK_UBUNTU_URL $UBUNTU_CODENAME main" > /etc/apt/sources.list.d/salt.list
+    ## DGM __apt_key_fetch "$SALTSTACK_UBUNTU_URL/SALT-PROJECT-GPG-PUBKEY-2023.gpg" || return 1
 
+    __fetch_url "/etc/apt/sources.list.d/salt.sources" "https://github.com/saltstack/salt-install-guide/releases/latest/download/salt.sources"
+    __apt_key_fetch "${HTTP_VAL}://${_REPO_URL}//api/security/keypair/SaltProjectKey/public" || return 1
```
Comment (karelfridrich) [inline bootstrap-salt.sh]: //api should be /api

**Comment by lburinov73** [comment]
Does not work on AlmaLinux 9

```
[root@example ~]# curl -s https://raw.githubusercontent.com/saltstack/salt-bootstrap/0899e72c34958da284c29fd71eeddb52cc5abe9f/bootstrap-salt.sh | sh
 *  INFO: Running version: 2024.11.01
 *  INFO: Executed by: shell pipe
 *  INFO: Command line: 'sh '
 *  WARN: Running the unstable version of bootstrap-salt.sh

 *  INFO: System Information:
 *  INFO:   CPU:          AuthenticAMD
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   5.14.0-427.40.1.el9_4.x86_64
 *  INFO:   Distribution: AlmaLinux 9.4

 *  INFO: Installing minion
 *  INFO: Found function install_almalinux_onedir_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_almalinux_onedir
 *  INFO: Found function install_almalinux_onedir_post
 *  INFO: Found function install_almalinux_restart_daemons
 *  INFO: Found function daemons_running_onedir
 *  INFO: Found function install_almalinux_check_services
 *  INFO: Running install_almalinux_onedir_deps()
main: line 4191: REPO_FILE: unbound variable

[root@example ~]# cat /etc/almalinux-release 
AlmaLinux release 9.4 (Seafoam Ocelot)
```

**Comment by joelpmichael** [comment]
Also not working on Debian 12 (after manually fixing the issue with `//api/security/keypair/SaltProjectKey/public`):

```
root@template:~# ./bootstrap-salt.sh 
 *  INFO: Running version: 2024.11.01
 *  INFO: Executed by: /bin/sh
 *  INFO: Command line: './bootstrap-salt.sh '
 *  WARN: Running the unstable version of bootstrap-salt.sh

 *  INFO: System Information:
 *  INFO:   CPU:          
 *  INFO:   CPU Arch:     aarch64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   6.1.21-v8+
 *  INFO:   Distribution: Debian 12

 *  INFO: Installing minion
 *  INFO: Found function install_debian_onedir_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_debian_onedir
 *  INFO: Found function install_debian_restart_daemons
 *  INFO: Found function daemons_running_onedir
 *  INFO: Found function install_debian_check_services
 *  INFO: Running install_debian_onedir_deps()
Hit:1 http://security.debian.org/debian-security bookworm-security InRelease
Hit:2 http://ftp.au.debian.org/debian bookworm InRelease
Hit:3 http://ftp.au.debian.org/debian bookworm-updates InRelease
Hit:4 http://archive.raspberrypi.org/debian bookworm InRelease
Hit:5 https://packages.broadcom.com/artifactory/saltproject-deb stable InRelease
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
procps is already the newest version (2:4.0.2-3).
pciutils is already the newest version (1:3.9.0-4).
python3-yaml is already the newest version (6.0-3+b2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
wget is already the newest version (1.21.3-1+b1).
gnupg2 is already the newest version (2.2.40-1.1).
apt-transport-https is already the newest version (2.6.1).
ca-certificates is already the newest version (20230311).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Hit:1 http://security.debian.org/debian-security bookworm-security InRelease
Hit:2 http://ftp.au.debian.org/debian bookworm InRelease
Hit:3 http://ftp.au.debian.org/debian bookworm-updates InRelease
Hit:4 http://archive.raspberrypi.org/debian bookworm InRelease
Hit:5 https://packages.broadcom.com/artifactory/saltproject-deb stable InRelease
Reading package lists...
./bootstrap-salt.sh: 3535: STABLE_REV: parameter not set
```

**Comment by tjyang** [comment]
> curl -s https://raw.githubusercontent.com/saltstack/salt-bootstrap/0899e72c34958da284c29fd71eeddb52cc5abe9f/bootstrap-salt.sh

- This STABLE_REV issue also applied to Ubuntu 24.04.
```
#sed  //api/ to be /api/ 
/root/bootstrap-salt.sh.11.01 stable 3006
<snipped>
/root/bootstrap-salt.sh.11.01: 3013: STABLE_REV: parameter not set
```
- workaround solution for bootstrap-salt.sh 11.01 that works on ubuntu 24.04.
```
STABLE_REV=3006 /root/bootstrap-salt.sh.11.01 stable
```

**Comment by djivey** [comment]
To get rhel based deployments to work with the new bootstrap. I had to make the following changes.

Note: I am using salt-cloud to deploy. 

Line: 4191 - I had to move 'REPO_FILE="/etc/yum.repos.d/salt.repo"' above the condition. due to the error above in https://github.com/saltstack/salt-bootstrap/pull/2028#issuecomment-2452828592

Then I had to remove the .repo in the `__fetch_url` because it was breaking the url "https://github.com/saltstack/salt-install-guide/releases/latest/download/salt.repo" by appending .repo to the end of it.

I don't know if that is the best way to fix it, but it worked for me.

**Comment by ketzacoatl** [comment]
> > curl -s https://raw.githubusercontent.com/saltstack/salt-bootstrap/0899e72c34958da284c29fd71eeddb52cc5abe9f/bootstrap-salt.sh
> 
>     * This STABLE_REV issue also applied to Ubuntu 24.04.
> 
> 
> ```
> #sed  //api/ to be /api/ 
> /root/bootstrap-salt.sh.11.01 stable 3006
> <snipped>
> /root/bootstrap-salt.sh.11.01: 3013: STABLE_REV: parameter not set
> ```
> 
>     * workaround solution for bootstrap-salt.sh 11.01 that works on ubuntu 24.04.
> 
> 
> ```
> STABLE_REV=3006 /root/bootstrap-salt.sh.11.01 stable
> ```

It seems to work on Ubuntu 22.04 LTS, but fail elsewhere at:
```
    amazon-ebs:  * ERROR: https://packages.broadcom.com/artifactory//api/security/keypair/SaltProjectKey/public failed to download to /tmp/salt-gpg-dnGWmUJR.pub
    amazon-ebs:  * ERROR: Failed to run install_ubuntu_onedir_deps()!!!
```

```
    amazon-ebs:  *  INFO: Running version: 2024.11.01
    amazon-ebs:  *  INFO: Executed by:
    amazon-ebs:  *  INFO: Command line: 'sh stable 3006'
    amazon-ebs:  *  WARN: Running the unstable version of bootstrap-salt.sh
==> amazon-ebs:     50K .......... .......... .......... .......... .......... 34% 1.81M 0s
    amazon-ebs:
    amazon-ebs:  *  INFO: System Information:
    amazon-ebs:  *  INFO:   CPU:          GenuineIntel
    amazon-ebs:  *  INFO:   CPU Arch:     x86_64
    amazon-ebs:  *  INFO:   OS Name:      Linux
    amazon-ebs:  *  INFO:   OS Version:   6.2.0-1012-aws
    amazon-ebs:  *  INFO:   Distribution: Ubuntu 22.04
```

File: bootstrap-salt.sh (hunk 1 above)
Comment (dmurphy18) [reply bootstrap-salt.sh]: done

**Comment by Foorack** [comment]
It also looks like this PR switches from using `salt.list` to `salt.sources` ?

File: .github/workflows/test-windows.yml [hunk 2]
```diff
@@ -47,65 +41,32 @@ jobs:
...
         run: |
-          env
-          bundle exec kitchen verify ${{ matrix.instance }}-${{ inputs.distro-slug }}
-          sleep 2
+          $env:Path = "$env:Path;C:\Program Files\Salt Project\Salt"
+          $env:Instance = ${{ matrix.instance }}
+          Write-Output "Path:"
+          Write-Output $env:Path
+          Write-Output "Environment Variables"
+          Get-ChildItem Env:
```
Comment (twangboy) [inline .github/workflows/test-windows.yml:65]: This is old Windows code

**Review by twangboy — APPROVED** [review]

File: .github/workflows/test-windows.yml (hunk 2 above)
Comment (dmurphy18) [reply .github/workflows/test-windows.yml:65]: Merge your windows stuff in after

**Comment by dmurphy18** [comment]
Note: problem with Photon OS, it ignores ```exclude``` in the repo file, hence always getting ```latest``` version of Salt.
Aslo shell script implements the Salt Install Guide, https://docs.saltproject.io/salt/install-guide/en/latest/ and hence only getting latest Major version of Salt , that is, specify 3006.8, you will get 3006.9 which is the latest for the 3006 branch.

**Pull request merged**